src/lookup-table/utils/dataTypeUtils.ts

```diff
@@ -0,0 +1,103 @@
+import { mapValues } from 'lodash'
+import { FieldMapType } from '../types'
+import { DateTime } from 'luxon'
+
+export const setDataTypes = (fieldMaps: FieldMapType[], rows: object[]) => {
+  // Determine type
+  const done: string[] = []
+  for (const row of rows) {
+    Object.entries(row).forEach(([key, value], index) => {
+      if (!done.includes(key)) {
+        const type = getType(value)
+        if (type !== null) {
+          fieldMaps[index].dataType = type
+          done.push(key)
+        }
+      }
+    })
+
+    if (done.length === fieldMaps.length) break
+  }
+
+  const fieldMap: { [key: string]: any } = fieldMaps.reduce((acc, curr) => {
+    const name = curr.fieldname
+    const type = curr.dataType
+    return { ...acc, [name]: type }
+  }, {})
+
+  // Now mutate rows based on those types
+  rows.forEach((row, index) => {
+    rows[index] = mapValues(row, (value, key) => {
+      if (key === 'id') return value
+      return convertType(value, fieldMap[key])
```

Comment:
All values coming from CSV files are strings, so this converts the actual data to the correct type (i.e. `"2"` becomes `2`, `"FALSE"` becomes `false` etc.